specs/bond-manager.md

```diff
@@ -0,0 +1,112 @@
+# Bond Manager Interface
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+- [The Bond Problem](#the-bond-problem)
+  - [Simple Bond](#simple-bond)
+  - [Variable Bond](#variable-bond)
+- [Contract Interface](#contract-interface)
+- [Bond Manager Implementation](#bond-manager-implementation)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+In the context of permissionless output proposals, bonds are a value that must
+be attached to an output proposal. In this case, the bond will be paid in ether.
+
+By requiring a bond to be posted with an output proposal, spam and invalid outputs
+are disincentivized. It's important to note that it is directly disincentivized
+because if the output is invalid, it will be seized by a set of challenge agents
+when the output is deleted. Thus, the bond acts as a form of economic security.
+
+Concretely, outputs will be permissionlessly proposed to the `L2OutputOracle` contract.
+When calling the propose function, the ether value is sent as the bond. This bond is
+then held by a bond manager contract. The bond manager contract is responsible for
+both the [dispute-games](./dispute-game.md) and the `L2OutputOracle` (further detailed
+in [proposals](./proposals.md)).
+
+The bond manager will need to handle bond logic for a variety of different
+[dispute-games](./dispute-game-interface.md). In the simplest "attestation" dispute game,
+bonds will not be required since the attestors are permissioned. But in more complex games,
+such as the fault dispute game, challengers and defenders perform a series of
+alternating onchain transactions requiring bonds at each step.
+
+## The Bond Problem
+
+At its core, the bond manager is straightforward - it escrows or holds ether and returns
+it at maturity or seized. But the uncertainty of introducing bonds lies in the
```

Comment:
At the moment, the other specs do not call for the bond manager returning the bond once the proposal reaches maturity (i.e. a `push`). The bond can be pulled from the bond manager and pushed to the entities that participate in providing attestor signatures to the attestation game upon resolution, though.

This opens up a bit of an interesting design decision where a caller of `OptimismPortal.finalizeWithdrawalTransaction` returns the bond on a matured output proposal (if it has not already been returned), but I think it makes more sense to go with @smartcontracts' initial idea from CDMX where the BondManager silo's liquid (withdrawable) funds from locked (bonded) funds.

Reply:
Ah yes, I didn't mean to intend a push here, but will make it explicit since you interpreted it as such :)